lua/lazyvim/plugins/extras/lang/csharp_ls.lua

```diff
@@ -0,0 +1,104 @@
+return {
+  recommended = function()
+    return LazyVim.extras.wants({
+      ft = { "cs", "vb" },
+      root = { "*.sln", "*.csproj", "omnisharp.json", "function.json" },
+    })
+  end,
+
+  { "Decodetalkers/csharpls-extended-lsp.nvim", lazy = true },
+
+  {
+    "nvim-treesitter/nvim-treesitter",
+    opts = { ensure_installed = { "c_sharp" } },
+  },
+  {
+    "nvimtools/none-ls.nvim",
+    optional = true,
+    opts = function(_, opts)
+      local nls = require("null-ls")
+      opts.sources = opts.sources or {}
+      table.insert(opts.sources, nls.builtins.formatting.csharpier)
+    end,
+  },
+  {
+    "stevearc/conform.nvim",
+    optional = true,
+    opts = {
+      formatters_by_ft = {
+        cs = { "csharpier" },
+      },
+      formatters = {
+        csharpier = {
+          command = "dotnet-csharpier",
+          args = { "--write-stdout" },
+        },
+      },
+    },
+  },
+  {
+    "williamboman/mason.nvim",
+    opts = { ensure_installed = { "csharpier", "netcoredbg", "csharp-language-server" } },
+  },
+  {
+    "neovim/nvim-lspconfig",
+    opts = {
+      servers = {
+        sharp_ls = {
```

Comment:
```suggestion
        csharp_ls = {
```
I believe this should be `csharp_ls` and with this you could also remove `csharp-language-server` from `mason.ensure_installed`, since when defining the LSP server in `nvim-lspconfig` spec it will also automatically download the LSP server.